app-modules/workflow/src/Filament/Blocks/CaseBlock.php

```diff
@@ -227,4 +229,45 @@ public static function type(): string
     {
         return 'workflow_case_details';
     }
+
+    /**
+     * @param array<string, mixed> $data
+     *
+     * @return array<string, mixed>
+     */
+    public function beforeCreate(array $data): array
+    {
+        return $this->transformAssignmentData($data);
+    }
+
+    /**
+     * @param array<string, mixed> $data
+     *
+     * @return array<string, mixed>
+     */
+    public function beforeUpdate(array $data): array
+    {
+        return $this->transformAssignmentData($data);
+    }
+
+    public function prepareForEdit(WorkflowDetails $details): void
+    {
+        if ($details instanceof WorkflowCaseDetails) {
```

Comment:
Is there a situation where this is `false`? If not, `assert()` is probably better